ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/RemoveUnexpectedArgumentsFix.fs

```diff
@@ -0,0 +1,25 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+
+type RemoveUnexpectedArgumentsFix(warning: NotAFunctionError) =
+    inherit FSharpQuickFixBase()
+        
+    let notAFunctionExpr = warning.NotAFunctionExpr
+    let unexpectedArgs = warning.UnexpectedArgs
+    
+    override x.Text = "Remove unexpected arguments"
+
+    override x.IsAvailable _ =
+        isValid notAFunctionExpr && not (notAFunctionExpr :? ILiteralExpr)
+
+    override x.ExecutePsiTransaction _ =
+        use writeCookie = WriteLockCookie.Create(notAFunctionExpr.IsPhysical())
+        for arg in unexpectedArgs do
+            let firstToDelete = getFirstMatchingNodeBefore (fun x -> x.IsWhitespaceToken()) arg
+            let lastToDelete = getLastMatchingNodeAfter (fun x -> x.IsWhitespaceToken()) arg
+            deleteChildRange firstToDelete lastToDelete

```

Comment:
You have to try to keep the tree in a correct form.

After these modifications you'll end up having prefixApp expressions that have no argument and other feature may not work on these tree nodes correctly. Since what you're doing is basically removing application of arguments you should actually replace the application nodes accordingly.

_Maybe_ it'll be easier to keep the applications in the highlighting instead of the args.